### PR TITLE
Refactor to improve IDE auto-completion support

### PR DESCRIPTION
**Pull request by adamnsch**
**NOTE:** It looks bigger than it really is. Mostly things have just been moved to different files/classes.

Thank you for your contribution to the Graph Data Science Client project.

Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md). 

Make sure:
- [x] You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
- [x] Your contribution is covered by tests

### CODE REVIEW

**Comment by adamnsch** [comment]
> Thank you for your contribution to the Graph Data Science Client project.
> 
> Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md).
> 
> Make sure:
> 
> * [ ]  You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
> * [ ]  Your contribution is covered by tests

It looks a lot bigger than it is. Mostly things have just been moved to different files :man_shrugging:

File: graphdatascience/graph/graph_alpha_proc_runner.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import os
```
Comment (adamnsch) [inline graphdatascience/graph/graph_alpha_proc_runner.py:1]: Moved here from `GraphProcRunner`

File: graphdatascience/graph/graph_beta_proc_runner.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+from typing import Any, List, Tuple, Union
```
Comment (adamnsch) [inline graphdatascience/graph/graph_beta_proc_runner.py:1]: Moved here from `GraphProcRunner`

File: graphdatascience/model/model_alpha_proc_runner.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+from typing import Any, Tuple
```
Comment (adamnsch) [inline graphdatascience/model/model_alpha_proc_runner.py:1]: Moved here from `ModelProcRunner`

File: graphdatascience/model/model_beta_proc_runner.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+from typing import Any, Optional
```
Comment (adamnsch) [inline graphdatascience/model/model_beta_proc_runner.py:1]: Moved here from `ModelProcRunner`

File: graphdatascience/model/model_resolver.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+from ..caller_base import CallerBase
```
Comment (adamnsch) [inline graphdatascience/model/model_resolver.py:1]: Moved here from `ModelProcRunner`

File: graphdatascience/pipeline/pipeline_beta_proc_runner.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
+from typing import Any, Optional
```
Comment (adamnsch) [inline graphdatascience/pipeline/pipeline_beta_proc_runner.py:1]: Moved here from `PipelineProcRunner`

File: graphdatascience/graph/graph_beta_proc_runner.py (hunk 2 above)
Comment (brs96) [reply graphdatascience/graph/graph_beta_proc_runner.py:1]: The methods here are also still in `graph_proc_runner`, should they be removed there too, like those in `graph_alpha_proc_runner`?

**Review by brs96 — APPROVED** [review]
Nice refactoring, 👍  Could you just check whether those methods in graph_proc_runner should be removed?
@Mats-SX what's the outcome of the API tiers? Will alpha/beta namespace stay?

**Comment by Mats-SX** [comment]
@brs96 Unfortunately, no decision has been made yet. I will notify you as soon as I can.

**Comment by adamnsch** [comment]
> @brs96 Unfortunately, no decision has been made yet. I will notify you as soon as I can.

According to Luke, if we do change it will take a while. So merging this

File: graphdatascience/graph/graph_beta_proc_runner.py (hunk 2 above)
Comment (adamnsch) [reply graphdatascience/graph/graph_beta_proc_runner.py:1]: Good catch! `generate` should not be in `GraphProcRunner`. The others need to be in both though

**Pull request merged**